content/en/docs/concepts/security/service-accounts.md

```diff
@@ -178,8 +178,8 @@ following methods:
   rotates the token before it expires.
 * [Service Account Token Secrets](/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token)
   (not recommended): You can mount service account tokens as Kubernetes
-  Secrets in Pods. These tokens don't expire and don't rotate. This method
-  is not recommended, especially at scale, because of the risks associated
+  Secrets in Pods and External applications. These tokens don't expire and don't rotate.
```

Comment:
This change doesn't make sense...

Reply:
Hi @tengqm these changes are according to the following [discussion](https://kubernetes.slack.com/archives/C0EN96KUY/p1676957078025019).
in docs, it is mentioned that the Service Account Token Secrets are not recommended for the pods but nothing is mentioned for the external service/applications in docs.
So just to make the docs clear I added the `external application` word here. Could suggest something more so that it will be clear that Service Account Token Secrets are also not recommended for external applications?

Reply:
> Hi @tengqm these changes are according to the following [discussion](https://kubernetes.slack.com/archives/C0EN96KUY/p1676957078025019).
> in docs, it is mentioned that the Service Account Token Secrets are not recommended for the pods but nothing is mentioned for the external service/applications in docs.
> So just to make the docs clear I added the external application word here. Could suggest something more so that it will be clear that Service Account Token Secrets are also not recommended for external applications?
> 
> Thanks

Hi @tengqm 
Your thoughts on this?

Reply:
What do you mean by mounting service account tokens into external applications?

Reply:
As you know that to communicate k8s, external applications use a Service account. Previously `Service Account Token Secrets` were used but now service account tokens would be obtained via TokenRequest API and stored as a projected volume in the pod, but we still use the Manually create a long-lived Service Account Token Secrets token but they are not recommended for the pods.
My concern is, can we still use long-lived Service Account Token Secrets for external applications? as per the discussion we can but they are also not recommended for it like pods.

Following use case for the external application mentioned in the [docs](https://kubernetes.io/docs/concepts/security/service-accounts/#use-cases): 
An external service needs to communicate with the Kubernetes API server. For example, authenticating to the cluster as part of a CI/CD pipeline.

line quote from the [discussion](https://kubernetes.slack.com/archives/C0EN96KUY/p1676971320852229?thread_ts=1676957078.025019&cid=C0EN96KUY):

> We could clarify that the problem with using long-term secrets is not where you use them but how long they are valid.

So just want to add something for external applications so that it will be more clear.

Reply:
I can vaguely imagine the use case. You have an application that runs outside of the Kubernetes cluster and this application is called "External application". This is where my first confusion came from. We may want to rephrase the sentence so that the meaning of "External application" is clear to the users.

Your application wants to authenticate to the Kubernetes API server and you choose to use ServiceAccount token. That is a valid use case IMO. You will use the ServiceAccount token stored in a Secret. This is also fine. What Kubernetes doesn't encourage to do is to mount that Secret into a projected volume. This has nothing to do with your "external application". That is where my second confusion came from.

The revised text says this "You can mount service account tokens as Kubernetes Secrets in Pods and External applications." ... This text is in accurate. It is causing more confusion rather than clarifying things as we wished.

Reply:
Hi, @tengqm  Thanks for taking a look at this.

> Your application wants to authenticate to the Kubernetes API server and you choose to use ServiceAccount token. That is a valid use case IMO. You will use the ServiceAccount token stored in a Secret. This is also fine. What Kubernetes doesn't encourage to do is to mount that Secret into a projected volume. This has nothing to do with your "external application". That is where my second confusion came from.

As you mentioned also we can use ServiceAccount token secrets((not automatically created for service account from k8s v1.24)) for External applications. So this thing I want to highlight in the docs like it is mentioned for the pods. 

> The revised text says this "You can mount service account tokens as Kubernetes Secrets in Pods and External applications." ... This text is in accurate. It is causing more confusion rather than clarifying things as we wished.

Ok, if it seems inaccurate, so can we mention a note or something you will suggest that clarify the things for external applications that  `Service Account Token Secret` (not automatically created for service account from k8s v1.24) is used for external applications/services but they are not recommended similar to pods.

Reply:
Right. "For applications running outside your Kubernetes cluster, you can use ServiceAccount token that is stored in a Secret. This is not recommended. You are encouraged to use the TokenRequest API instead."

Reply:
Thanks @tengqm I will update it and add a note here.

Reply:
Hi, @tengqm I have addressed the comments PTAL!